### ISSUE TITLE
"paddle.summary" counts the number of parameters incorrectly

### ISSUE THREAD

**Issue by Wong4j** [PaddlePaddle/Paddle#40057]
Thank you for contributing to PaddlePaddle.
Before submitting the issue, you could search issue in the github in case that there was a similar issue submitted or resolved before.
If there is no solution,please provide us with the following details :
**System information**
-PaddlePaddle version: v2.2
-CPU: None
-GPU: V100 CUDA11.4
-OS Platform: Ubuntu 20.04
-Python version: 3.8
Note: You can get most of the information by running [summary_env.py](https://github.com/PaddlePaddle/Paddle/blob/develop/tools/summary_env.py).       

**To Reproduce**
Use "paddle.summary" to print the number of parameters of ResNet50 and BERT model.    

**Describe your current behavior**
I have been trying to use some tools to check the parameters of each layer of the RN50 and BERT model. When I compared the results of PyToch and Paddle, I found that "paddle.summary" counts the number of parameters incorrectly.     

**Code to reproduce the issue**
For ResNet50, run the script:
```python
import paddle
from paddle.vision.models import resnet50
model = resnet50()
paddle.summary(model, (1,3,224,224))
```

For BERT:
Insert codes here [PaddleNLP-BERT](https://github.com/PaddlePaddle/PaddleNLP/blob/4a9106570b04029b892739a420ba64c8a03da3f6/examples/language_model/bert/static/run_pretrain.py#L295) (after the 295th line) with following codes, then launch bert pretraining following PaddleNLP-BERT README :
```python
paddle.summary(model, (64, 128), dtypes="int64")
import sys
sys.exit()
```

**Other info / logs**
For ResNet50, "summary" miscalculated the trainable and non-trained parameters of BatchNorm, so the total number trainable and non-trainable parameters is also wrong: 
![image](https://user-images.githubusercontent.com/21985950/156286640-1715446c-d530-4aec-a9de-c54da5576463.png)
The correct number of total trainable parameters is 25557032, which is also verified in PyTorch.

For BERT, "summary" correctly counts the number of parameters in each layer of the model, but gives a wrong number of total parameters as below:
![image](https://user-images.githubusercontent.com/21985950/156287917-73752748-d0ac-4d0e-b018-cf1ed846861f.png)

As we know, the number of parameter of BERT-base is 110M. With "BertPretraningHeads", the total number of parameters would reach about 130M. But "paddle.summary" gives "490M". I guess some numbers are added up repeatedly.

**Comment by xingjing1** [comment]
@LielinJiang

**Comment by wangna11BD** [comment]
Hello, the problem you reported is currently a bug in paddle.summary, we will arrange to fix it, thank you for your feedback.

**Comment by mingxu1067** [comment]
There are still some different between Paddle and PyTorch.
Paddle:
> * code
```python
import paddle
from paddle.vision.models import resnet50
model = resnet50()
paddle.summary(model, (1,3,224,224))
```
> * result
```
Total params: 25,610,152
Trainable params: 25,557,032
Non-trainable params: 53,120
```

PyTorch:
> * code
```python
from torchvision import models
from torchsummary import summary
torch_model = models.resnet50()
summary(torch_model)
```
> * result
```
Total params: 25,557,032
Trainable params: 25,557,032
Non-trainable params: 0
```

The question here is where do those `Non-trainable params` (53,120) come from?

**Comment by wangna11BD** [comment]
nn.BatchNorm2d has 4 parameters: 'weight', 'bias', 'mean' and 'variance', where 'mean' and 'variance' are generally non-trainable parameters. Pytorch only counts the parameters of 'weight' and 'bias', while paddle counts 4 parameters, including the trainable parameters 'weight' and 'bias', and the non-trainable parameters 'mean' and 'variance'.

**Comment by mingxu1067** [comment]
Verified to close
